### PR TITLE
Optimize 'string.Compare' call

### LINKED ISSUE #5578: Optimize 'string.Compare' call
They were found using Roslynator
https://github.com/JosefPihrt/Roslynator/blob/master/docs/analyzers/RCS1235.md

### PR DESCRIPTION
**Pull request by elachlan**
Fixes #5578

### CODE REVIEW

File: src/Build.OM.UnitTests/Construction/ProjectRootElement_Tests.cs [hunk 1]
```diff
@@ -532,7 +532,7 @@ public void ValidXmlXmlReaderCache()
         public void LoadCommonTargets()
         {
             ProjectCollection projectCollection = new ProjectCollection();
-            string toolsPath = projectCollection.Toolsets.Where(toolset => (string.Compare(toolset.ToolsVersion, ObjectModelHelpers.MSBuildDefaultToolsVersion, StringComparison.OrdinalIgnoreCase) == 0)).First().ToolsPath;
+            string toolsPath = projectCollection.Toolsets.Where(toolset => (string.Equals(toolset.ToolsVersion, ObjectModelHelpers.MSBuildDefaultToolsVersion, StringComparison.OrdinalIgnoreCase))).First().ToolsPath;
```
Comment (Forgind) [inline src/Build.OM.UnitTests/Construction/ProjectRootElement_Tests.cs:535]: nit:
extra parens

File: src/Deprecated/Engine/Resources/Constants.cs [hunk 2]
```diff
@@ -45,16 +45,16 @@ internal static bool IsReservedProperty(string property)
...
-                    (String.Compare(property, projectFile, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, projectExtension, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, projectFullPath, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, projectName, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, binPath, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, toolsPath, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, projectDefaultTargets, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, programFiles32, StringComparison.OrdinalIgnoreCase) == 0) ||
-                    (String.Compare(property, assemblyVersion, StringComparison.OrdinalIgnoreCase) == 0)
+                    (String.Equals(property, projectDirectory, StringComparison.OrdinalIgnoreCase)) ||
```
Comment (Forgind) [inline src/Deprecated/Engine/Resources/Constants.cs:48]: lot of extra parens

File: src/Deprecated/Engine/Introspector/TargetInProgressState.cs [hunk 3]
```diff
@@ -582,7 +582,7 @@ public override bool Equals(object obj)
                 if (other != null)
                 {
                     if (other.projectId == projectId && other.nodeId == nodeId &&
-                        (String.Compare(other.name, name, StringComparison.OrdinalIgnoreCase) == 0))
+                        (String.Equals(other.name, name, StringComparison.OrdinalIgnoreCase)))
```
Comment (Forgind) [inline src/Deprecated/Engine/Introspector/TargetInProgressState.cs:585]: extra parens

File: src/Deprecated/Engine/Engine/Project.cs [hunk 4]
```diff
@@ -4175,7 +4175,7 @@ private XmlDocument LoadImportedProject(Import import)
             // also prevents the same file from being imported twice, even it it's not a
             // circular dependency, but that's fine -- no good reason to do that anyway.
             if ((this.imports[import.EvaluatedProjectPath] != null) ||
-                (string.Compare(this.FullFileName, import.EvaluatedProjectPath, StringComparison.OrdinalIgnoreCase) == 0))
+                (string.Equals(this.FullFileName, import.EvaluatedProjectPath, StringComparison.OrdinalIgnoreCase)))
```
Comment (Forgind) [inline src/Deprecated/Engine/Engine/Project.cs:4178]: extra parens

**Comment by elachlan** [comment]
> I gave up on marking all the extra parentheses pretty fast this time. I know it all came in before you touched it, but it does bother me slightly. Same with sometimes using String. and sometimes using string.

I think I'll do a separate issue to remove extra parenthesis and add in ones that should be there. That way the review is specific.

**Pull request merged**